state-chain/custom-rpc/src/lib.rs

```diff
@@ -1086,6 +1086,35 @@ where
 	) -> Self {
 		Self { rpc_backend: CustomRpcBackend::new(client, backend, executor) }
 	}
+
+	pub fn into_block_update<Data>(
+		&self,
+		data: RpcResult<Data>,
+		at: Option<Hash>,
+	) -> RpcResult<BlockUpdate<Data>> {
+		let block_hash = self.rpc_backend.unwrap_or_best(at);
+		let block_number = self
```

Comment:
This is repeating/duplicating some work:
- when we call `with_runtime_api` we already have the hash
- the header also contains the hash
- the header might also contain the height? 

I would like us to think a little harder about how to integrate the block wrapper nicely, without minimsl futher backend queries.